src/scripts/main.js

```diff
@@ -1,5 +1,29 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = wall.querySelector('.spider');
+
+  if (e.target === wall) {
+    const coords = wall.getBoundingClientRect();
+
+    spider.style.left = e.clientX - coords.left - spider.offsetWidth / 2 + 'px';
+    spider.style.top = e.clientY - coords.top - spider.offsetHeight / 2 + 'px';
```

Comment:
The calculation for positioning the spider is correct, but ensure that 'spider.offsetWidth' and 'spider.offsetHeight' are defined and return valid numbers. If the spider element is not yet rendered or has no size, this could lead to unexpected behavior.